### PR TITLE
fix Range._get_output_shape_from_returnn

### PR DESCRIPTION
**Pull request by vieting**
In `Range._get_output_shape_from_returnn`, if `limit` is a `Tensor` we currently set the `dim_tag` for the `SizeValue` in `torch_shape`, but not the `originating_tensor`. This might be problematic as can be seen in the testcase that I added

### CODE REVIEW

File: pytorch_to_returnn/torch/nn/modules/operator.py [hunk 1]
```diff
@@ -31,18 +31,16 @@ def create_returnn_layer_dict(self, limit, start, delta, dtype, sparse=False) ->
...
       assert limit.is_defined
-      limit_dim = None
+      size = SizeValue((int(limit.numpy()) - int(start)) // int(delta))
       if limit.returnn_naming_entry.is_size_value is not None:
-        limit_dim = limit.returnn_naming_entry.is_size_value.dim_tag
-      limit = limit.numpy()
-    from .shape import SizeValue
-    size = SizeValue((int(limit) - int(start)) // int(delta))
-    if limit_dim:
-      size.dim_tag = (limit_dim - int(start)) // int(delta)
```
Comment (albertz) [inline pytorch_to_returnn/torch/nn/modules/operator.py:45]: Why did you change the old code so much?
Wouldn't it have been enough to just add `size.originating_tensor = ...` here?

File: pytorch_to_returnn/torch/nn/modules/operator.py [hunk 2]
```diff
@@ -31,18 +31,16 @@ def create_returnn_layer_dict(self, limit, start, delta, dtype, sparse=False) ->
...
+        size.dim_tag = limit.returnn_naming_entry.is_size_value.dim_tag
```
Comment (albertz) [inline pytorch_to_returnn/torch/nn/modules/operator.py:40]: This is wrong. It should be `(limit_dim - int(start)) // int(delta)` like it was before.

File: pytorch_to_returnn/torch/nn/modules/operator.py (hunk 1 above)
Comment (vieting) [reply pytorch_to_returnn/torch/nn/modules/operator.py:45]: I found this 
```
limit_dim = None
...
if limit_dim:
...
```
unnecessarily long and harder to read, especially if we do the same for `originating_tensor` now. But if you prefer, I can also add `originating_tensor` in the same way.

Comment (albertz) [reply pytorch_to_returnn/torch/nn/modules/operator.py:45]: Why so complicated? Instead of `limit_dim`, you could also directly keep the `size_value`.

File: pytorch_to_returnn/torch/nn/modules/operator.py [hunk 3]
```diff
@@ -31,18 +31,16 @@ def create_returnn_layer_dict(self, limit, start, delta, dtype, sparse=False) ->
...
-        limit_dim = limit.returnn_naming_entry.is_size_value.dim_tag
-      limit = limit.numpy()
-    from .shape import SizeValue
-    size = SizeValue((int(limit) - int(start)) // int(delta))
-    if limit_dim:
-      size.dim_tag = (limit_dim - int(start)) // int(delta)
+        size.dim_tag = (limit.returnn_naming_entry.is_size_value.dim_tag - int(start)) // int(delta)
+        size.originating_tensor = limit.returnn_naming_entry.is_size_value.originating_tensor
+    else:
+      size = SizeValue((int(limit) - int(start)) // int(delta))
```
Comment (albertz) [inline pytorch_to_returnn/torch/nn/modules/operator.py:43]: This code is redundant. I think you should not repeat the code `size = SizeValue(...)` twice.

File: pytorch_to_returnn/torch/nn/modules/operator.py [hunk 4]
```diff
@@ -31,18 +31,16 @@ def create_returnn_layer_dict(self, limit, start, delta, dtype, sparse=False) ->
...
-      limit_dim = None
+      size = SizeValue((int(limit.numpy()) - int(start)) // int(delta))
       if limit.returnn_naming_entry.is_size_value is not None:
-        limit_dim = limit.returnn_naming_entry.is_size_value.dim_tag
-      limit = limit.numpy()
-    from .shape import SizeValue
-    size = SizeValue((int(limit) - int(start)) // int(delta))
-    if limit_dim:
-      size.dim_tag = (limit_dim - int(start)) // int(delta)
+        size.dim_tag = (limit.returnn_naming_entry.is_size_value.dim_tag - int(start)) // int(delta)
```
Comment (albertz) [inline pytorch_to_returnn/torch/nn/modules/operator.py:40]: Do not access always `limit.returnn_naming_entry.is_size_value`. This makes it hard to read. First do sth like
```
limit_size = limit.returnn_naming_entry.is_size_value
```

**Review by albertz — APPROVED** [review]
